forte/data/base_store.py

```diff
@@ -0,0 +1,160 @@
+# Copyright 2019 The Forte Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from abc import abstractmethod
+from typing import (
+    List,
+    Type,
+    Union,
+    Iterator,
+)
+
+from forte.data.ontology.core import EntryType
+
+__all__ = ["BaseStore"]
+
+
+class BaseStore:
+    r"""The base class which will be used by :class:
+    `~forte.data.data_store.DataStore`."""
+
+    # pylint: disable=too-many-public-methods
```

Comment:
are we having `too-many-public-methods` in this class? if not, don't disable this